stringdtype/stringdtype/src/dtype.c

```diff
@@ -66,34 +62,65 @@ string_discover_descriptor_from_pyobject(PyArray_DTypeMeta *NPY_UNUSED(cls),
     return ret;
 }
 
+static PyObject *
+get_value(PyObject *scalar)
+{
+    PyObject *ret_bytes = NULL;
+    PyTypeObject *scalar_type = Py_TYPE(scalar);
+    // FIXME: handle bytes too
```

Comment:
Maybe numpy bytes only?  Normal bytes shouldn't be necessary, Python 3 doesn't do it.  NumPy bytes are just weird because they serve the dual purpose of an ascii/latin1 string.